Documentation/design-docs/unloadability.md

```diff
@@ -0,0 +1,111 @@
+﻿# `AssemblyLoadContext` unloadability
+## Goals
+* Provide a building block for the AppModel team to implement unloadable plug-ins
+* Users can load an assembly and its dependencies into an unloadable `AssemblyLoadContext`. 
+* There can be multiple unloadable `AssemblyLoadContext` instances coexisting
+* Users can unload the `AssemblyLoadContext` when there are no outside hard references to the types and instance of types from the assemblies loaded into the context and the assemblies themselves.
+* The unload can be initiated eagerly using explicit `Unload` method or it can happen after all references to the `AssemblyLoadContext` and all the assemblies in it are gone.
+## Non-goals
+* Forceful unload using thread abort or similar measures to cause threads to stop running with frames of code from the context to be unloaded on the call stack.
+## General scenarios
+Based on various discussions and feedback on github, the following general scenarios were often mentioned as use cases for unloadability.
+* Loading assemblies for one-off introspection. 
+* Plugin scenarios when dynamic plugin loading and unloading is required.
+* Dynamically compiling, running and then flushing code. Useful for web sites, scripting engines, etc.
+## Real-world scenarios
+A couple of real-world scenarios were analyzed to assess feasibility of the `AssemblyLoadContext` unloading as a replacement of `AppDomain` unloading and also for possible new usages of the `AssemblyLoadContext` unloading.
+### Roslyn
+Roslyn has historically used AppDomains for executing different tests that can have assemblies with the same names but different identity or even the same identity, but different contents. `AppDomain` per test was very slow. So, to speed it up, they made them reusable based on the assemblies loaded into them.
+* In .NET Core, they use `AssemblyLoadContext` without unloading, one context per test works fine. 
+* Due to the missing plugin model, especially the dependency resolving issues, they use Mono.
```

Comment:
This is not correct. Roslyn is testing on all .NET Framework, .NET Core and Mono.